js/components/common/react_modal.js.jsx

```diff
@@ -979,4 +979,36 @@ function convertTimeToString(time) {
   return ((time === 12.5 ? 12 : Math.floor(time)) % 12) + ":30" + meridiem
 }
 
-export { CourseModal, MapModal, FocusModal }
+/**
+ * For use in Generate - a warning modal for when the user enters one or more invalid courses
+ *
+ * Props:
+ *  - message: The message to be displayed
+ *  - onClose: the function to be invoked when the close-button is clicked (or `esc`)
+ *  - isOpen: boolean that determines whether the modal should be visible or not
+ */
+function ErrorMessage({ title, message, onClose, isOpen }) {
+  return (
+    <ReactModal
+      className="error-modal-class"
+      overlayClassName="error-overlay"
+      isOpen={isOpen}
+      ariaHideApp={false}
+      onRequestClose={onClose}
+      contentLabel="Pop-up warning for invalid courses"
+    >
+      <div className="modal-header">
+        {title}
+        <div className="button-container">
```

Comment:
I like that you're following an existing style here. However, I think buttons with icons/symbols are most appropriate for the positioning here. An "X" could be okay.

I _also_ liked the button you had at the bottom of the modal originally. Perhaps add that (in addition to the "X") with a label that says "Okay".